Sprint-3/quote-generator/quotes.js

```diff
@@ -491,3 +491,57 @@ const quotes = [
 ];
 
 // call pickFromArray with the quotes array to check you get a random quote
+
+function displayRandomQuote() {
+  const quoteElement = document.getElementById("quote");
+  const authorElement = document.getElementById("author");
+  const maninElement = document.querySelector("main");
+
+  // Fade out current content
+  quoteElement.style.opacity = "0";
+  authorElement.style.opacity = "0";
+  maninElement.style.opacity = "0";
+
+  setTimeout(() => {
+    const randomQuote = pickFromArray(quotes);
+
+    quoteElement.textContent = randomQuote.quote;
+    authorElement.textContent = `_ ${randomQuote.author}`;
+
+    quoteElement.style.opacity = "1";
+    authorElement.style.opacity = "1";
+    maninElement.style.opacity = "1";
+  }, 300);
+}
+
+let timer = null;
+
+document.getElementById("toggle-timer").addEventListener("change", (event) => {
+  const statusSpan = document.getElementById("status");
+
+  if (event.target.checked) {
+    statusSpan.textContent = "Auto-Play is :ON";
+
+    timer = setTimeout(() => {
+      timer = setInterval(displayRandomQuote, 60000);
+      displayRandomQuote();
+    }, 6000);
```

Comment:
`setTimeout()` is redundant. Lines 525-528 can be replaced by:
```
      // Schedule a function to be called every 6s
      timer = setInterval(displayRandomQuote, 6000); 
```

Reply:
Thank you for reviewing my code and providing your feedback!

Regarding your comment, I wanted to clarify that I followed the guideline mentioned in line 24:
// DO NOT modify this array, otherwise the tests may break!

I did not make any changes to the code below this point, ensuring the array and other parts remain intact.

Additionally, line 493 includes a comment:

// call pickFromArray with the quotes array to check you get a random quote

This suggested to me that it's appropriate to call pickFromArray with the quotes array. Based on this, I added my code accordingly without modifying the prohibited sections.
---------------------------------------------------------------------------------------
-The reason I placed the setInterval() call inside the setTimeout() function is to ensure that the quote does not change immediately when the checkbox is clicked. Instead, a timer starts after 6 seconds, with the next quote appearing at 60-second intervals.

This approach avoids giving the user the impression that the checkbox is behaving like a button to change the quote immediately. It maintains a consistent timing experience for the user, aligning with the intended behavior of the feature.


Please let me know if you have additional suggestions or need further clarification!